File: c/gemini/p84.c
#include <microhttpd.h>
#include <stdio.h>
#include <string.h>
#include <yaml.h>

#define PORT 5000

enum MHD_Result answer_to_connection(void *cls, struct MHD_Connection *connection,
                      const char *url, const char *method, const char *version,
                      const char *upload_data, size_t *upload_data_size, void **con_cls) {
    
    const char *payload = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "payload");
    const char *response_text;
    
    if (!payload) {
        response_text = "Error: Missing payload parameter";
    } else if (strstr(payload, "type: Create")) { // Simplified check for C example
        response_text = "Error: Type 'Create' is not allowed";
    } else if (strstr(payload, "type: Update") && strstr(payload, "name: UnsafeDeserialization")) {
        response_text = "Match found: UnsafeDeserialization detected.";
    } else {
        response_text = "Payload processed.";
    }

    struct MHD_Response *response = MHD_create_response_from_buffer(strlen(response_text), (void*)response_text, MHD_RESPMEM_PERSISTENT);
    enum MHD_Result ret = MHD_queue_response(connection, MHD_HTTP_OK, response);
    MHD_destroy_response(response);
    return ret;
}

int main() {
    struct MHD_Daemon *daemon = MHD_start_daemon(MHD_USE_INTERNAL_POLLING_THREAD, PORT, NULL, NULL, &answer_to_connection, NULL, MHD_OPTION_END);
    if (!daemon) return 1;
    getchar();
    MHD_stop_daemon(daemon);
    return 0;
}

File: java/gemini/p84.java
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.*;
import org.yaml.snakeyaml.Yaml;
import java.util.Map;

@SpringBootApplication
@RestController
public class YamlApplication {

    public static void main(String[] args) {
        SpringApplication.run(YamlApplication.class, args);
    }

    @GetMapping("/")
    public String handleRequest(@RequestParam(required = false) String payload) {
        if (payload == null || payload.isEmpty()) {
            return "Error: Missing payload parameter";
        }

        try {
            Yaml yaml = new Yaml();
            Map<String, Object> data = yaml.load(payload);

            if ("Create".equals(data.get("type"))) {
                return "Error: Type 'Create' is not allowed";
            }

            if ("Update".equals(data.get("type")) && 
                "UnsafeDeserialization".equals(data.get("name"))) {
                return "Match found: UnsafeDeserialization detected.";
            }

            return "Payload processed successfully.";
        } catch (Exception e) {
            return "Error: Invalid YAML format";
        }
    }
}